ch_pipeline/hfb/containers.py

```diff
@@ -51,7 +52,32 @@ def nsample(self) -> memh5.MemDataset:
             raise KeyError("Dataset 'nsample' not initialised.")
 
 
-class HFBData(RawContainer, FreqContainer, HFBContainer):
+class HFBBeamContainer(HFBContainer):
+    """A pipeline container for HFB data with a beam axis.
+
+    This works like a normal :class:`HFBContainer` container, but already has a beam
+    axis defined, and specific properties for dealing with beams.
+    """
+
+    _axes = ("beam",)
+
+    @property
+    def beam(self) -> np.ndarray:
+        """The beam indices associated with each entry of the beam axis."""
+        return self.index_map["beam"]
+
+    @cached_property
+    def beam_ew(self) -> np.ndarray:
+        """The unique EW-beam indices (i.e., from 0 to 3) in the beam axis."""
+        return np.unique(self.beam // 256)
+
+    @cached_property
+    def beam_ns(self) -> np.ndarray:
+        """The unique NS-beam indices (i.e., from 0 to 256) in the beam axis."""
```

Comment:
NS-beam indices run from 0 to 255 (and not from 0 to 256).

Reply:
Fixed.